// @mtpjr88/lotr-sdk/dist/api/index.js
import axios from "axios";
var Headers;
(function (Headers) {
    Headers["Authorization"] = "Authorization";
})(Headers || (Headers = {}));
class Api {
    errorHandler() {
        throw new Error('client must be initialized, try envoking ApiUtil.init()');
    }
    constructor(token) {
        this.init(token);
    }
    init(token) {
        this.client = axios.create({
            baseURL: 'https://the-one-api.dev/v2/',
            timeout: 1000,
            headers: { [Headers.Authorization]: `Bearer ${token}` }
        });
        return this;
    }
    async get(url, args) {
        var _a;
        return (_a = (await this.client.get(url, args))) === null || _a === void 0 ? void 0 : _a.data;
    }
    async post(url, args) {
        var _a;
        return (_a = (await this.client.post(url, args))) === null || _a === void 0 ? void 0 : _a.data;
    }
    async put(url, args) {
        var _a;
        return (_a = (await this.client.put(url, args))) === null || _a === void 0 ? void 0 : _a.data;
    }
    async destroy(url, args) {
        return (await this.client.delete(url, args)).data;
    }
}
export default Api;
