// @mtpjr88/lotr-sdk/dist/main/index.js
import Api from "../api";
var Routes;
(function (Routes) {
    Routes["BOOK"] = "/book";
    Routes["MOVIE"] = "/movie";
    Routes["CHARACTER"] = "/character";
    Routes["QUOTE"] = "/quote";
    Routes["CHAPTER"] = "/chapter";
})(Routes || (Routes = {}));
class LotrSDK extends Api {
    constructor(token) {
        super(token);
        this.getAllBooks = async () => {
            return this.get(Routes.BOOK);
        };
        this.getBookById = async (id) => {
            return this.get(`${Routes.BOOK}/${id}`);
        };
        this.getAllMovies = async () => {
            return this.get(Routes.MOVIE);
        };
        this.getMovieById = async (id) => {
            return this.get(`${Routes.MOVIE}/${id}`);
        };
        this.getMovieQuotesById = async (id) => {
            const route = `${Routes.MOVIE}/${id}${Routes.QUOTE}`;
            return this.get(route);
        };
        this.getAllCharacters = async () => {
            return this.get(Routes.CHARACTER);
        };
        this.getCharacter = async (id) => {
            return this.get(`${Routes.CHARACTER}/${id}`);
        };
        this.getCharacterQuotesById = async (id) => {
            return this.get(`${Routes.CHARACTER}/${id}${Routes.QUOTE}`);
        };
        this.getAllQuotes = async (id) => {
            return this.get(Routes.QUOTE);
        };
        this.getQuoteById = async (id) => {
            return this.get(`${Routes.QUOTE}/${id}`);
        };
        this.getAllBookChapters = async (id) => {
            return this.get(Routes.CHAPTER);
        };
        this.getBookChapter = async (id) => {
            return this.get(`${Routes.CHAPTER}/${id}`);
        };
    }
}
export default LotrSDK;
